### PR TITLE
Fix the area feature

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change

This is the first of two PR:s that will introduce the possibilty to use targets on CallService. This PR adds the new HassClient that has the new functionality plus fixes the Area handling on entities that has area assigned directly without device. It may break existing test code since new HassClient has made more properties `init` instead of `set`. Next PR will implement the target functionality.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code compiles without warnings (code quality chek)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/Daemon/NetDaemon.Daemon/Daemon/EntityStateManagement.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ public async Task RefreshAsync()
                                 entity_id = entityId,
                                 state,
                                 attributes
-                            }, waitForResponse)
+                            }, null, waitForResponse)
```
Comment (helto4real) [inline src/Daemon/NetDaemon.Daemon/Daemon/EntityStateManagement.cs:67]: Just adding null where ever target can be added for now. Next PR will add this to CallService too.

File: src/Fakes/NetDaemon.Fakes/HassClientMock.cs [hunk 2]
```diff
@@ -60,7 +62,7 @@ public HassClientMock()
             SetupGet(x => x.States).Returns(FakeStates);
 
             Setup(x => x.GetAllStates(It.IsAny<CancellationToken>()))
-                .ReturnsAsync(() => FakeStates.Values);
+                .ReturnsAsync(() => new ReadOnlyCollection<HassState>(FakeStates.Values.ToList()));
```
Comment (helto4real) [inline src/Fakes/NetDaemon.Fakes/HassClientMock.cs:65]: Kinda flunky but for tests this is needed to convert to ReadOnlyCollection<T>

File: src/Fakes/NetDaemon.Fakes/HassClientMock.cs [hunk 3]
```diff
@@ -25,17 +27,17 @@ public class HassClientMock : Mock<IHassClient>
         /// <summary>
         ///     Fake areas in HassClient
         /// </summary>
-        public HassAreas Areas { get; } = new();
+        public ICollection<HassArea> Areas { get; } = new List<HassArea>();
```
Comment (FrankBakkerNl) [inline src/Fakes/NetDaemon.Fakes/HassClientMock.cs:30]: Not an IReadonlyCollection?

File: src/Fakes/NetDaemon.Fakes/HassClientMock.cs (hunk 2 above)
Comment (FrankBakkerNl) [reply src/Fakes/NetDaemon.Fakes/HassClientMock.cs:65]: Isnt there an extensiomethod like AsReadonly() or smtng like that?

**Review by FrankBakkerNl — APPROVED** [review]

File: src/Fakes/NetDaemon.Fakes/HassClientMock.cs (hunk 3 above)
Comment (helto4real) [reply src/Fakes/NetDaemon.Fakes/HassClientMock.cs:30]: It needs to be writable so we can use it to mock. It is converted into read only later.

File: src/Fakes/NetDaemon.Fakes/HassClientMock.cs (hunk 2 above)
Comment (helto4real) [reply src/Fakes/NetDaemon.Fakes/HassClientMock.cs:65]: Sorry missed this comment and I merged it. Will change it next PR

**Pull request merged**